### PR TITLE
Decorators : Decorate the Person class

### PR DESCRIPTION
**Pull request by kensteph**
Hi there 👋!
In this PR I : 
✅ Create a Nameable class and correct_name method.
✅ Make Person class inherits from Nameable.
✅ Implement the method correct_name.
✅ Create a Decorator class and its constructor.
✅ Implement the correct_name method.
✅ Create CapitalizeDecorator class and implement the correct_name method.
✅ Create TrimmerDecorator class and implement the correct_name method.
✅ Fix linter errors.

### CODE REVIEW

File: decorator.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+require_relative './nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
```
Comment (zieeco) [inline decorator.rb:6]: - Hey, let's also add the `attr_accessor :nameable` for the nameable so this could be accessed.

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @kensteph ,





Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice decoration. ✔️
- Descriptive PR. ✔️
- Good commit messages :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._




## Optional suggestions
- N/A


**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by kensteph** [comment]
Thank you @zieeco for your review.

**Review by mukhammadrizooff — APPROVED** [review]
Hi @kensteph, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good  implementations ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**